File: ios/Earwyrm/Core/Models/Collection.swift
import Foundation

struct Collection: Codable, Identifiable, Hashable {
    let id: UUID
    let userId: UUID
    let name: String
    let description: String?
    let color: String?
    let isSmart: Bool?
    let smartTag: String?
    let createdAt: Date
    let updatedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case name
        case description
        case color
        case isSmart = "is_smart"
        case smartTag = "smart_tag"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

struct CollectionInsert: Encodable {
    let userId: UUID
    let name: String
    let description: String?
    let color: String?
    let isSmart: Bool
    let smartTag: String?

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case name
        case description
        case color
        case isSmart = "is_smart"
        case smartTag = "smart_tag"
    }
}

File: android/app/src/main/java/com/earwyrm/app/core/model/Collection.kt
package com.earwyrm.app.core.model

import kotlinx.datetime.Instant
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EarwyrmCollection(
    val id: String, @SerialName("user_id") val userId: String, val name: String,
    val description: String? = null, val color: String? = null,
    @SerialName("is_smart") val isSmart: Boolean? = null,
    @SerialName("smart_tag") val smartTag: String? = null,
    @SerialName("created_at") val createdAt: Instant,
    @SerialName("updated_at") val updatedAt: Instant? = null
)

@Serializable
data class CollectionInsert(
    @SerialName("user_id") val userId: String, val name: String,
    val description: String? = null, val color: String? = null,
    @SerialName("is_smart") val isSmart: Boolean? = null,
    @SerialName("smart_tag") val smartTag: String? = null
)

@Serializable
data class LyricCollection(
    val id: String, @SerialName("lyric_id") val lyricId: String,
    @SerialName("collection_id") val collectionId: String,
    @SerialName("added_at") val addedAt: Instant
)

@Serializable
data class LyricCollectionInsert(
    @SerialName("lyric_id") val lyricId: String,
    @SerialName("collection_id") val collectionId: String
)
